docs/pages/application-access/troubleshooting-apps.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Troubleshooting Application Access
+description: Describes common issues and solutions for access to applications.
+---
+
+This section describes common issues that you might encounter in managing access to applications
+with Teleport and how to work around or resolve them.
+
+## Unable to access applications
+
+The most common issues that prevent access to applications involve Cross-Site Request Forgery (CSRF) or
+Cross-Origin Resource Sharing (CORS) errors.
+
+Cross-Site Request Forgery is a type of attack that uses an authenticated user's identity to perform operations
+without the user's knowledge. For example, a CSRF attack might transfer funds, change passwords, or make purchases
+by issuing a forged request with the user's credentials. Browsers and applications have checks to prevent these
+types of attacks. However, the checks can also block legitimate requests under certain conditions.
+
+### Symptom
+
+If your browser can't create a secure cookie or can't authorize your login from a previously-created secure 
+cookie, you might see the following error:
+
+Invalid or missing CSRF token
+
+This error can be caused by ad-blocking or script-blocking extensions or by the browser itself.
+For access to applications through Teleport, you might see this error with most often with 
+applications—like Grafana and ArgoCD—that use WebSockets extensively and in browsers where 
+cross-site scripting restrictions require traffic to be explicitly allowed.
+
+Issues with Cross-Site Request Forgery (CSRF) or Cross-Origin Resource Sharing (CORS) usually 
+result in a loss of application functionality, errors in the application itself indicating that 
+traffic isn't being permitted, or application logs that indicate CORS or CSRF errors. 
+
+### Solution
+
+In most cases, you can fix these types of issues by adding explicit `rewrite` settings for the Origin and Host headers 
+in the Teleport configuration file for each application.
+
+To fix CSRF or CORS issues:
+
+1. Open the `/etc/teleport.yaml` file in a text editor.
```

Comment:
We should be explicit that this is the teleport.yaml file for the app service(s) that are proxying the app in question.

Reply:
@webvictim, so would the fix be the same but just in a different file? For example (remove [ ] and add the rewrite.headers section):
47   apps:
       - name: grafana
         uri: http://localhost:3000/
         public_addr: [grafana.teleport.example.com](http://grafana.teleport.example.com/)
         rewrite:
             headers:
                Origin: "https://grafana.teleport.example.com/" # should be the Teleport application subdomain prepended with "https://"
                Host: "[grafana.teleport.example.com](http://grafana.teleport.example.com/)" # should be the Teleport application subdomain itself